sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/converters/ChatRetentionPolicyConverter.java

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.chat.implementation.converters;
+
+import com.azure.communication.chat.models.ChatRetentionPolicy;
+import com.azure.communication.chat.models.NoneRetentionPolicy;
+import com.azure.communication.chat.models.ThreadCreationDateRetentionPolicy;
+
+/**
+ * A converter between {@link com.azure.communication.chat.implementation.models.ChatRetentionPolicy} and
+ * {@link ChatRetentionPolicy}.
+ */
+public final class ChatRetentionPolicyConverter {
+    /**
+     * Maps from the implementation {@link com.azure.communication.chat.implementation.models.ChatRetentionPolicy}
+     * to the public {@link ChatRetentionPolicy} model.
+     *
+     * @param obj the implementation model to convert
+     * @return a {@link ChatRetentionPolicy} instance, or null if the input is null or not a compatible type
+     */
+    public static ChatRetentionPolicy convertFromImpl(Object obj) {
+        if (!(obj instanceof com.azure.communication.chat.implementation.models.ChatRetentionPolicy)) {
+            return null;
```

Comment:
When could `obj` not be a `ChatRetentionPolicy`? Would it not make more sense to throw here?

Reply:
When we create a thread on older versions of APIs that not supporting retention policy. The response would not contain a retention policy field. I think we should return null to not breaking requests to previous APIs.